### ISSUE TITLE
pointer generator model starts overfitting

### ISSUE THREAD

**Issue by priyanks179** [ishalyminov/copy_seq2seq#2]
I am trying to train pointer generator network . After training for 10k iterations it starts overfitting.
Any suggestions on why this might be happening. coverage_loss is not used.
Note- 1==50 iterations. coverage_loss is not used.
![adagrad 90k](https://user-images.githubusercontent.com/40527782/54050004-259f2580-4204-11e9-97b4-48dde2317156.png)

**Comment by ishalyminov** [comment]
Hard to say without knowing your exact model and data.. (I had the same problem when I was training it on bAbI and tested on Switchboard which are respectively a synthetic and a natural dataset.)
But if you experience overfitting, normally you would want to decrease the learning rate/reduce the model's size/apply standard regularization techniques e.g. L2-regularization, dropout, batch norm, multitasking.

**Comment by priyanks179** [comment]
Sir i am giving you the link to repo of the model . it would be great if you could see it coz i am working on this thing for a month now but i am not able to figure out the reason for overfitting even after including dropout and Batch-norm.
Link : https://github.com/priyanks179/pointer-gen-network

**Comment by ishalyminov** [comment]
Well, it doesn't look like this repo is used in your code - not sure why you created the issue here..
A general advice for fighting overfitting is to either further simplify your model (if you tried dropout and batch norm, you still have layer sizes and learning rate to tweak) or getting more training data by e.g. synthetic data augmentation.
I can't provide you with any specific comments regarding your code, sorry.
Good luck!
